apps/aemon/src/aemon_metrics.erl

```diff
@@ -46,42 +80,35 @@ publisher_balance(Balance) ->
 publisher_post_tx(Action) ->
     update([publisher, post_tx, Action], 1).
 
-%% ttl
-
 queue_size(Count) ->
     update([publisher, queue, size], Count).
 
 ttl_expired(Count) ->
     update([publisher, queue, ttl_expired], Count).
 
-%% on start metric creation
-
-create(on_chain) ->
-    create([forks, micro], counter),
-    create([confirmation, delay], histogram),
+block_propagation_time(Type, Time) when Time < 1 ->
+    lager:debug("Ignoring metric update of block_propagation_time.~p, Time = ~p", [Type, Time]),
     ok;
-create(gen_stats) ->
-    create([gen_stats, tx, total], histogram),
-    create([gen_stats, tx, monitoring], histogram),
-    create([gen_stats, microblocks, total], histogram),
-    ok;
-create(publisher) ->
-    create([publisher, balance], gauge),
-    create([publisher, post_tx, success], counter),
-    create([publisher, post_tx, max_adjustment], counter),
-    create([publisher, post_tx, nonce_too_low], counter),
-    create([publisher, post_tx, nonce_too_high], counter),
+block_propagation_time(Type, Time) when Type =:= key; Type =:= micro ->
+    update([block, propagation_time, Type], Time).
+
+block_time_since_prev(Type, Time) when Time < 1 ->
+    lager:debug("Ignoring metric update of block_time_since_prev.~p, Time = ~p", [Type, Time]),
     ok;
-create(ttl) ->
-    create([publisher, queue, size], histogram),
-    create([publisher, queue, ttl_expired], histogram),
-    ok.
+block_time_since_prev(Type, Time) when Type =:= key; Type =:= micro ->
+    update([block, time_since_prev, Type], Time).
 
-%% internals
+chain_top_difficulty(Difficulty) ->
+    update([chain, top, difficulty], Difficulty).
+
+%% ==================================================================
+%% internal functions
 
 create(Name, Type) ->
-    exometer:ensure(metric(Name), Type, []),
-    ok.
+    create(Name, Type, []).
+
+create(Name, Type, Opts) ->
+    ok = exometer:ensure(metric(Name), Type, Opts).
```

Comment:
Are you sure you ant to crash here in case we cannot create the metric?

Reply:
I agree we could just log a failure. I just found the previous approach of ignoring it entirely out of place.